libs/remix-lib/src/execution/txListener.ts

```diff
@@ -164,26 +160,33 @@ export class TxListener {
     this._isListening = false
   }
 
-  _startListenOnNetwork () {
+  async _startListenOnNetwork () {
+    let lastSeenBlock = this.executionContext.lastBlock?.number
     this._loopId = setInterval(() => {
       const currentLoopId = this._loopId
-      this.executionContext.web3().eth.getBlockNumber((error, blockNumber) => {
-        if (this._loopId === null) return
-        if (error) return console.log(error)
-        if (currentLoopId === this._loopId && blockNumber > this.lastBlock) {
-          let current = this.lastBlock + 1
-          this.lastBlock = blockNumber
-          while (blockNumber >= current) {
-            try {
-              this._manageBlock(current)
-            } catch (e) {
-              console.log(e)
-            }
-            current++
+      if (this._loopId === null) return
+      if (!lastSeenBlock) {
+        lastSeenBlock = this.executionContext.lastBlock?.number // trying to resynchronize
+        console.log('listen on blocks, resynchronising')
+        return
+      }
+      const current = this.executionContext.lastBlock?.number
+      if (!current) {
+        console.error(new Error('no last block found'))
```

Comment:
does this mean we would see a console message on each loop when no block is found? Is this not something to show in the terminal itself? as a warning... or something.

Reply:
this will be show in the browser terminal. not sure we should put it in the remix terminal, don't want to spam it...

Reply:
but you can definetly make it a log or warnng